pkg/db/db_test.go

```diff
@@ -200,4 +210,74 @@ func TestVendor_GetActiveCertificates(t *testing.T) {
 
 func TestOrganization_GetActiveCertificates(t *testing.T) {
 	assert.Len(t, Organization{}.GetActiveCertificates(), 0)
+}
+
+func TestOrganization_HasKey(t *testing.T) {
+	t.Run("ok - not found", func(t *testing.T) {
+		keyAsJWK := generateJWK()
+		o := Organization{}
+		hasKey, err := o.HasKey(keyAsJWK, time.Now())
+		assert.NoError(t, err)
+		assert.False(t, hasKey)
+	})
+	t.Run("ok - match: deprecated public key", func(t *testing.T) {
+		key, _ := rsa.GenerateKey(rand.Reader, 2048)
+		keyAsJWK, _ := jwk.New(key)
+		pem, _ := pkg.PublicKeyToPem(&key.PublicKey)
```

Comment:
Why generate a new one instead of using the above defined `deprecatedPublicKey`?

Reply:
I prefer tests to generate their input data since that generally leads to more maintainable more useful (since test data differs each run) tests, so I'll even replace `deprecatedPublicKey` by a generated key :-)